projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -58,9 +66,20 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
   protected emitNullOnDestroy = true;
   protected emitInitialValueOnInit = true;
 
+  // private controlKeys: (keyof FormInterface)[] = [];
+
   private subscription: Subscription | undefined = undefined;
 
   // can't define them directly
+  // private _getFormControls(): Controls<FormInterface> {
+  //   const controls: Controls<FormInterface> = this.getFormControls();
+
+  //   this.controlKeys = (Object.keys(controls) as unknown) as (keyof FormInterface)[];
+
+  //   // console.log('this.controlKeys', this.controlKeys);
+
+  //   return controls;
+  // }
```

Comment:
@zakhenry I thought I could set the control keys once and then access them later but later I **always** get an empty array and I have absolutely no clue why :man_shrugging: 

It's currently working but it bothers me a bit to have to loop over the object properties every time the form is being updated when we know properties are static...

Reply:
I'll prob have to debug with you in person, but at the console.log point, is the `const controls` also empty?

Reply:
Nvm, got too curious so trying myself

Reply:
Explained in slack, but for posterity - just move the `private controlKeys: (keyof FormInterface)[] = [];` declaration to above the `public formGroup: TypedFormGroup<FormInterface> = new FormGroup(` so the constructor ordering is correct

Reply:
Nice one! Thanks :)